commercetools-internal-processors/src/main/java/io/sphere/sdk/annotations/processors/generators/BaseAbstractGenerator.java

```diff
@@ -237,26 +238,33 @@ protected AnnotationSpec createJsonPropertyAnnotation(final String jsonName) {
     }
 
     protected MethodSpec createFactoryMethod(final FactoryMethod factoryMethod, final List<PropertyGenModel> properties, final ClassName returnType) {
-        final Set<String> factoryParameterNames = Stream.of(factoryMethod.parameterNames()).collect(Collectors.toCollection(LinkedHashSet::new));
+        return createFactoryMethod(properties,returnType,factoryMethod.parameterNames(),factoryMethod.methodName(),factoryMethod.useLowercaseBooleans());
+    }
+
+    protected MethodSpec createFactoryMethod(final List<PropertyGenModel> properties, final ClassName returnType, final String[] parametersName,
```

Comment:
If would prefer if we would change `parametersName` to `parameterNames`. And for consistency it would be better to use `List<String>` instead of `String[]`.